Active crashes on sync when the cache manager is not configured with a default server resource

Steps to reproduce:
1. Start the active.
2. Create a cache manager without a default server resource configured.
3. Start the passive that triggers the sync.

Active will crash with the following exception

```
com.tc.exception.TCRuntimeException: Uncaught exception in stage
	at com.tc.async.impl.StageImpl$WorkerThread.run(StageImpl.java:211)
Caused by: java.lang.RuntimeException: org.terracotta.exception.EntityUserException: Entity: org.ehcache.clustered.client.internal.EhcacheClientEntity:my-application exception in user code: Default server resource cannot be null
	at com.tc.objectserver.entity.ManagedEntityImpl.invoke(ManagedEntityImpl.java:403)
	at com.tc.objectserver.entity.ManagedEntityImpl.lambda$sync$69(ManagedEntityImpl.java:756)
	at com.tc.objectserver.entity.ManagedEntityImpl$$Lambda$58/626438439.run(Unknown Source)
	at com.tc.objectserver.entity.RequestProcessor$EntityRequest.invoke(RequestProcessor.java:153)
	at com.tc.objectserver.entity.RequestProcessor$EntityRequest.run(RequestProcessor.java:145)
	at com.tc.objectserver.entity.RequestProcessorHandler.handleEvent(RequestProcessorHandler.java:27)
	at com.tc.objectserver.entity.RequestProcessorHandler.handleEvent(RequestProcessorHandler.java:23)
	at com.tc.async.impl.StageQueueImpl$HandledContext.runWithHandler(StageQueueImpl.java:502)
	at com.tc.async.impl.StageImpl$WorkerThread.run(StageImpl.java:197)
Caused by: org.terracotta.exception.EntityUserException: Entity: org.ehcache.clustered.client.internal.EhcacheClientEntity:my-application exception in user code: Default server resource cannot be null
	at com.tc.objectserver.entity.ManagedEntityImpl.invoke(ManagedEntityImpl.java:401)
	... 8 more
Caused by: java.lang.NullPointerException: Default server resource cannot be null
	at org.ehcache.clustered.common.ServerSideConfiguration.<init>(ServerSideConfiguration.java:50)
	at org.ehcache.clustered.server.EhcacheActiveEntity.synchronizeKeyToPassive(EhcacheActiveEntity.java:344)
	at com.tc.objectserver.entity.ManagedEntityImpl.performSync(ManagedEntityImpl.java:592)
	at com.tc.objectserver.entity.ManagedEntityImpl.invoke(ManagedEntityImpl.java:383)
	... 8 more
```

--- a/clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java
+++ b/clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java
@@ -340,8 +340,12 @@ class EhcacheActiveEntity implements ActiveServerEntity<EhcacheEntityMessage, Eh
   public void synchronizeKeyToPassive(PassiveSynchronizationChannel<EhcacheEntityMessage> syncChannel, int concurrencyKey) {
     LOGGER.info("Sync started for concurrency key {}.", concurrencyKey);
     if (concurrencyKey == DEFAULT_KEY) {
-      ServerSideConfiguration configuration =
-        new ServerSideConfiguration(ehcacheStateService.getDefaultServerResource(), ehcacheStateService.getSharedResourcePools());
+      ServerSideConfiguration configuration;
+      if (ehcacheStateService.getDefaultServerResource() == null) {
+        configuration = new ServerSideConfiguration(ehcacheStateService.getSharedResourcePools());
+      } else {
+        configuration = new ServerSideConfiguration(ehcacheStateService.getDefaultServerResource(), ehcacheStateService.getSharedResourcePools());
+      }
 
       Map<String, ServerStoreConfiguration> storeConfigs = new HashMap<>();
       for (String storeName : ehcacheStateService.getStores()) {

--- a/clustered/server/src/test/java/org/ehcache/clustered/server/EhcacheActiveEntityTest.java
+++ b/clustered/server/src/test/java/org/ehcache/clustered/server/EhcacheActiveEntityTest.java
@@ -2653,6 +2653,25 @@ public class EhcacheActiveEntityTest {
   }
 
   @Test
+  public void testSyncToPassiveWithoutDefaultServerResource() throws Exception {
+    final OffHeapIdentifierRegistry registry = new OffHeapIdentifierRegistry();
+    registry.addResource("serverResource1", 32, MemoryUnit.MEGABYTES);
+    registry.addResource("serverResource2", 32, MemoryUnit.MEGABYTES);
+
+    final EhcacheActiveEntity activeEntity = new EhcacheActiveEntity(registry, ENTITY_ID, DEFAULT_MAPPER);
+    ClientDescriptor client = new TestClientDescriptor();
+    activeEntity.connected(client);
+
+    ServerSideConfiguration serverSideConfiguration = new ServerSideConfigBuilder().build();
+
+    activeEntity.invoke(client,
+      MESSAGE_FACTORY.configureStoreManager(serverSideConfiguration));
+
+    PassiveSynchronizationChannel syncChannel = mock(PassiveSynchronizationChannel.class);
+    activeEntity.synchronizeKeyToPassive(syncChannel, 1);
+  }
+
+  @Test
   public void testLoadExistingRecoversInflightInvalidationsForEventualCache() {
     final OffHeapIdentifierRegistry registry = new OffHeapIdentifierRegistry();
     registry.addResource("serverResource1", 8, MemoryUnit.MEGABYTES);

